### PR TITLE
Inject a kernel-compat tester for kube-proxy test

### PR DESCRIPTION
**Pull request by thockin**
Obviates #19584 I think

### CODE REVIEW

File: pkg/proxy/iptables/proxier.go [hunk 1]
```diff
@@ -68,12 +68,18 @@ type IptablesVersioner interface {
 	GetVersion() (string, error)
 }
 
+// KernelCompatTester tests whether the required kernel capabilities are
+// present to run th eiptables proxier.
```
Comment (smarterclayton) [inline pkg/proxy/iptables/proxier.go]: nit: the iptables proxier

**Comment by smarterclayton** [comment]
One type, works on my machine

File: pkg/proxy/iptables/proxier.go (hunk 1 above)
Comment (thockin) [inline pkg/proxy/iptables/proxier.go]: Fixed

**Comment by smarterclayton** [comment]
LGTM

**Comment by thockin** [comment]
@k8s-bot test this please

**Comment by thockin** [comment]
@k8s-bot test this issue: #IGNORE

**Comment by thockin** [comment]
@k8s-bot unit test this issue: #IGNORE

**Pull request merged**